bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/SmartThingsApiService.java

```diff
@@ -0,0 +1,987 @@
+                                updateState(CHANNEL, tvChannel);
+                                String tvChannelName = d.getTvChannelName();
+                                logger.info("{}: SSE Got TV Channel Name: {}", host, tvChannelName);
+                                updateState(CHANNEL_NAME, tvChannelName);
+                            }
+                            String Power = d.getSwitch();
+                            if (!Power.isBlank()) {
+                                logger.info("{}: SSE Got TV Power: {}", host, Power);
+                                if ("on".equals(Power)) {
+                                    // handler.putOnline(); // ignore on event for now
+                                } else {
+                                    // handler.setOffline(); // ignore off event for now
+                                }
+                            }
+                        }, () -> logger.warn("{}: SSE Received NULL data", host));
+                    } catch (JsonSyntaxException e) {
+                        logger.warn("{}: SmartThingsApiService: Error ({}) in message: {}", host, e.getMessage(),
+                                rawData);
+                    }
+                    break;
+                default:
+                    logger.trace("{}: SSE not handling event: {}", host, event);
+                    break;
+            }
+        });
+    }
+
+    private boolean updateDeviceID(TvValues.Items item) {
+        this.deviceId = item.getDeviceId();
+        logger.info("{}: found {} device, adding device id {}", host, item.getName(), deviceId);
+        handler.putConfig(SMARTTHINGS_DEVICEID, deviceId);
+        prevUpdate = 0;
+        return true;
+    }
+
+    public boolean fetchdata() {
+        if (System.currentTimeMillis() >= prevUpdate + RATE_LIMIT) {
+            if (deviceId.isBlank()) {
+                tvInfo = fetchTVProperties(DEVICES);
+                boolean found = false;
+                if (tvInfo.isPresent()) {
+                    TvValues t = tvInfo.get();
+                    switch (t.getItems().length) {
+                        case 0:
+                        case 1:
+                            logger.info("{}: No devices found - please add your TV to the Smartthings app", host);
+                            break;
+                        case 2:
+                            found = Arrays.asList(t.getItems()).stream().filter(a -> "Samsung TV".equals(a.getName()))
+                                    .map(a -> updateDeviceID(a)).findFirst().orElse(false);
+                            break;
+                        default:
+                            logger.info("{}: No device Id selected, please enter one of the following:", host);
+                            Arrays.asList(t.getItems()).stream().forEach(a -> logger.info("{}: '{}' : {}({})", host,
+                                    a.getDeviceId(), a.getName(), a.getLabel()));
```

Comment:
I changed it to warn. If there are multiple TV's defined in smarthings, the binding attempts to configure the right one automatically, but if it is impossible to tell (because there are lots of different Samsung TV's, and they all report differently), you end up here.